include/swift/Frontend/Frontend.h

```diff
@@ -447,6 +450,13 @@ class CompilerInvocation {
 /// times on a single CompilerInstance is not permitted.
 class CompilerInstance {
   CompilerInvocation Invocation;
+
+  /// CAS Instances.
+  /// This needs to be declared before SourceMgr because when using CASFS,
+  /// the file buffer provided by CAS needs to outlive the SourceMgr.
+  std::shared_ptr<llvm::cas::ObjectStore> CAS;
+  std::shared_ptr<llvm::cas::ActionCache> Cache;
```

Comment:
Nit: Could we give this a more-descriptive name, please?

Reply:
I rename `Cache` -> `ResultCache`. `CAS` kept as it is since it is a common name for all the ObjectStore in llvm/clang